modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/SdkComponentsMinMuleVersionUtils.java

```diff
@@ -0,0 +1,655 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java.utils;
+
+import org.mule.metadata.api.model.ArrayType;
+import org.mule.runtime.api.connection.ConnectionProvider;
+import org.mule.runtime.api.meta.MuleVersion;
+import org.mule.runtime.extension.api.annotation.Configurations;
+import org.mule.runtime.extension.api.annotation.param.Config;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.annotation.param.ParameterGroup;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.source.Source;
+import org.mule.runtime.extension.api.runtime.streaming.PagingProvider;
+import org.mule.runtime.module.extension.api.loader.java.type.ConfigurationElement;
+import org.mule.runtime.module.extension.api.loader.java.type.ConnectionProviderElement;
+import org.mule.runtime.module.extension.api.loader.java.type.ExtensionElement;
+import org.mule.runtime.module.extension.api.loader.java.type.ExtensionParameter;
+import org.mule.runtime.module.extension.api.loader.java.type.FieldElement;
+import org.mule.runtime.module.extension.api.loader.java.type.FunctionElement;
+import org.mule.runtime.module.extension.api.loader.java.type.MethodElement;
+import org.mule.runtime.module.extension.api.loader.java.type.OperationContainerElement;
+import org.mule.runtime.module.extension.api.loader.java.type.OperationElement;
+import org.mule.runtime.module.extension.api.loader.java.type.SourceElement;
+import org.mule.runtime.module.extension.api.loader.java.type.Type;
+import org.mule.runtime.module.extension.api.loader.java.type.TypeGeneric;
+import org.mule.runtime.module.extension.api.loader.java.type.WithAnnotations;
+import org.mule.sdk.api.annotation.DoNotEnforceMinMuleVersion;
+import org.mule.sdk.api.annotation.Extension;
+import org.mule.sdk.api.annotation.MinMuleVersion;
+
+import javax.inject.Inject;
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Utils class to create {@link SdkComponent}s from {@link Type}s.
+ *
+ * @since 4.5
+ */
+public final class SdkComponentsMinMuleVersionUtils {
+
+  private static final MuleVersion SDK_EXTENSION_ANNOTATION_MIN_MULE_VERSION =
+      new MuleVersion(Extension.class.getAnnotation(MinMuleVersion.class).value());
+  public static final MuleVersion FIRST_MULE_VERSION = new MuleVersion("4.1.1");
```

Comment:
This actually looks like a constant that belongs someplace else as this is a fact not exclusive to this class domain